### PR TITLE
v2.4.0 release notes: STM32 and shields additions

### PR DESCRIPTION
**Pull request by erwango**
Update V2.4.0 release notes with :
- STM32 changes
- shields additions

### CODE REVIEW

File: doc/releases/release-notes-2.4.rst [hunk 1]
```diff
@@ -206,18 +218,23 @@ Drivers and Sensors
...
+  * STM32: Various changes including Flash latency wait states computation,
+    configuration option additions for H7 series, and fixes on F0/F3 PREDIV1
+    support
 
 * Console
 
 
 * Counter
 
+  * STM32: Add support on F0/F2 series
```
Comment (MaureenHelm) [inline doc/releases/release-notes-2.4.rst:235]:
```suggestion
  * STM32: Added support on F0/F2 series
```

File: doc/releases/release-notes-2.4.rst [hunk 2]
```diff
@@ -235,14 +254,16 @@ Drivers and Sensors
 
 * Entropy
 
+  * STM32: Add support for ISR mode. Add support on F7/H7/L0 series
```
Comment (MaureenHelm) [inline doc/releases/release-notes-2.4.rst:262]:
```suggestion
  * STM32: Added support for ISR mode. Added support on F7/H7/L0 series
```

File: doc/releases/release-notes-2.4.rst [hunk 3]
```diff
@@ -275,6 +296,8 @@ Drivers and Sensors
     to be synthesized to test driver behavior.  See
     :option:`CONFIG_I2C_EMUL`.
 
+  * STM32: V1: Reset i2c device on read/write error
+  * STM32: V2: Add dtrs configurable Timing option
```
Comment (MaureenHelm) [inline doc/releases/release-notes-2.4.rst:305]:
```suggestion
  * STM32: V2: Added dtrs configurable Timing option
```

File: doc/releases/release-notes-2.4.rst [hunk 4]
```diff
@@ -321,6 +344,7 @@ Drivers and Sensors
 
 * PWM
 
+  * STM32: Refactor using Cube LL API
```
Comment (MaureenHelm) [inline doc/releases/release-notes-2.4.rst:352]:
```suggestion
  * STM32: Refactored using Cube LL API
```

**Review by MaureenHelm — CHANGES_REQUESTED** [review]
thanks @erwango

**Comment by erwango** [comment]
Thanks for the review @MaureenHelm, all fixed now!

**Review by MaureenHelm — APPROVED** [review]

File: doc/releases/release-notes-2.4.rst [hunk 5]
```diff
@@ -173,23 +173,40 @@ Boards & SoC Support
 * Added support for these SoC series:
 
 
+* Made these changes in other SoC series
```
Comment (mbolivar-nordic) [inline doc/releases/release-notes-2.4.rst:176]:
```suggestion
* Made these changes in other SoC series:
```

**Review by mbolivar-nordic — APPROVED** [review]

**Comment by MaureenHelm** [comment]
Please rebase

File: doc/releases/release-notes-2.4.rst (hunk 5 above)
Comment (erwango) [reply doc/releases/release-notes-2.4.rst]: Fixed, txs for review.

**Pull request merged**